src/Multicall3.sol

```diff
@@ -78,34 +111,71 @@ contract Multicall3 {
         }
     }
 
+    /// @notice Aggregate calls with a msg value
+    /// @notice Reverts if msg.value is less than the sum of the call values
+    /// @param calls An array of Call3Value structs
+    /// @return returnData An array of Result structs
+    function aggregate3Value(Call3Value[] calldata calls) public payable returns (Result[] memory returnData) {
+        uint256 valAccumulator;
+        returnData = new Result[](calls.length);
+        uint256 length = calls.length;
+        Call3Value calldata call;
+        for (uint256 i = 0; i < length;) {
+            Result memory result = returnData[i];
+            call = calls[i];
+            valAccumulator += call.value;
```

Comment:
Probably could do unchecked here since it's impossible send enough ETH to overflow a uint256

Reply:
Ah good point

Reply:
Explicitly, there isn't enough `wei` to overflow a uint256

Reply:
Haha love the comment 😂